### PR TITLE
BoundingFrustum docs

### PR DESCRIPTION
**Pull request by Shqrdx**
Fully documented BoundingFrustum class.
-All public things are now documented
-All private fields are automatically renamed to private form with "_"(using resharper so it absolutly correct)
-Added Intersects overloads with rays(I dont know really how to get result differs from null and 0.0 so I leave this question opened). I also implemented few checks in Bounding.cs.
-PlaneCount is now opened const integer because why not ?
-ToString rewritted to more easy form.

### CODE REVIEW

File: MonoGame.Framework/BoundingFrustum.cs [hunk 1]
```diff
@@ -4,103 +4,178 @@
 
 using System;
 using System.Diagnostics;
-using System.Text;
 
 namespace Microsoft.Xna.Framework
 {
+    /// <summary>
+    /// Defines a frustum for intersection operations.
```
Comment (tomspilman) [inline MonoGame.Framework/BoundingFrustum.cs]: I would try to get a little better definition about what a frustum is here and how it is build from a matrix.  Not something huge, but something a little more.

File: MonoGame.Framework/BoundingFrustum.cs [hunk 2]
```diff
@@ -4,103 +4,178 @@
...
     public class BoundingFrustum : IEquatable<BoundingFrustum>
     {
         #region Private Fields
 
-        private Matrix matrix;
-        private readonly Vector3[] corners = new Vector3[CornerCount];
-        private readonly Plane[] planes = new Plane[PlaneCount];
+        private Matrix _matrix;
+        private readonly Vector3[] _corners = new Vector3[CornerCount];
+        private readonly Plane[] _planes = new Plane[PlaneCount];
```
Comment (tomspilman) [inline MonoGame.Framework/BoundingFrustum.cs]: I know how tempting it is to do these cleanups...  I really hate members without the leading `_` myself.  However you *really* should avoid doing this.  We do not want "cleanup" of code.

Go ahead and leave this in now, but please avoid it in the future.

File: MonoGame.Framework/BoundingFrustum.cs [hunk 3]
```diff
@@ -4,103 +4,178 @@
...
-        #endregion Private Fields
+        #endregion
 
         #region Public Fields
-        public const int CornerCount = 8;
-        #endregion
 
-        #region Public Constructors
+        /// <summary>
+        /// Plane count(6).
```
Comment (tomspilman) [inline MonoGame.Framework/BoundingFrustum.cs]: ```The number of planes in the frustum.```

Remove the number too.

File: MonoGame.Framework/BoundingFrustum.cs [hunk 4]
```diff
@@ -4,103 +4,178 @@
...
-        public BoundingFrustum(Matrix value)
-        {
-            this.matrix = value;
-            this.CreatePlanes();
-            this.CreateCorners();
-        }
-
-        #endregion Public Constructors
+        /// <summary>
+        /// Corner count(8).
```
Comment (tomspilman) [inline MonoGame.Framework/BoundingFrustum.cs]: ```The number of corner points in the frustum.```

Remove the number too.

File: MonoGame.Framework/BoundingFrustum.cs [hunk 5]
```diff
@@ -4,103 +4,178 @@
...
+                this._matrix = value;
                 this.CreatePlanes();    // FIXME: The odds are the planes will be used a lot more often than the matrix
-            	this.CreateCorners();   // is updated, so this should help performance. I hope ;)
-			}
+                this.CreateCorners();   // is updated, so this should help performance. I hope ;)
+            }
         }
 
+        /// <summary>
+        /// Gets the near plane of this frustum.
```
Comment (tomspilman) [inline MonoGame.Framework/BoundingFrustum.cs]: use `of the frustum`.

File: MonoGame.Framework/BoundingFrustum.cs (hunk 1 above)
Comment (Shqrdx) [inline MonoGame.Framework/BoundingFrustum.cs]: Please define it and I implement it

File: MonoGame.Framework/BoundingFrustum.cs (hunk 2 above)
Comment (Shqrdx) [inline MonoGame.Framework/BoundingFrustum.cs]: Ok, I forgot it a bit, so sorry... You will call automatically cleanup system ?

**Comment by Shqrdx** [comment]
@tomspilman It is ok for merging now ?

**Comment by Shqrdx** [comment]
@tomspilman Please recheck.

**Comment by tomspilman** [comment]
Merging.

**Pull request merged**